assets/queries/dockerfile/copy_from_without_from_alias_defined_previously/query.rego

```diff
@@ -0,0 +1,31 @@
+package Cx
+
+CxPolicy [ result ] {
+
+  resource := input.document[i].command[name][_]
+  resource.Cmd == "copy"
+  
+  contains(resource.Flags[x],"--from=")
+  aux_split := split(resource.Flags[x], "=")
+  
+  not isPreviousAlias(resource.StartLine, aux_split[1])
+
+  result := {
+                "documentId": 		  input.document[i].id,
+                "searchKey":        sprintf("FROM={{%s}}.{{%s}}", [name, resource.Original]),
+                "issueType":		    "IncorrectValue",
+                "keyExpectedValue": "COPY --from references a previous defined FROM alias",
+                "keyActualValue": 	"COPY --from does not reference a previous defined FROM alias"
+              }
+}
+
+isPreviousAlias(startLine, currentAlias) = allow {
+  resource := input.document[i].command[name][_]
+  resource.Cmd == "from"
+  previousAlias := resource.Value[2]
+  
+  previousAlias == currentAlias
+  resource.EndLine < startLine
+  
+  allow = true
+}
```

Comment:
COPY command with the flag '--from=0' shouldn't fail.
The COPY '--from=0' line copies just the built artifact from the previous stage into this new stage. The Go SDK and any intermediate artifacts are left behind, and not saved in the final image.